frame/post/src/lib.rs

```diff
@@ -0,0 +1,169 @@
+pub mod weights;
+
+use sp_std::prelude::*;
+use sp_runtime::{RuntimeDebug, SaturatedConversion, traits::Saturating};
+
+use frame_support::{decl_module, decl_event, decl_storage, decl_error, ensure};
+use frame_support::{
+	traits::{Currency, ReservableCurrency, Get},
+};
+use frame_system::ensure_signed;
+use codec::{Encode, Decode};
+pub use weights::WeightInfo;
+
+pub trait Trait: frame_system::Trait {
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+	/// Balances
+	type Currency: ReservableCurrency<Self::AccountId>;
+	/// The minimum deposit that needs to be placed for a post.
+	type MinDeposit: Get<BalanceOf<Self>>;
+	/// The amount per byte users need to deposit.
+	type ByteDeposit: Get<BalanceOf<Self>>;
+	/// The max length for a topic.
+	type MaxTopicLength: Get<usize>;
+	/// The max length for a post.
+	type MaxPostLength: Get<usize>;
+	/// Weight information for extrinsics in this pallet.
+	type WeightInfo: WeightInfo;
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, Default)]
+pub struct Post<Balance, BlockNumber> {
+	pub deposit: Balance,
+	pub block_number: BlockNumber,
+	pub post: Vec<u8>,
+}
+
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug)]
+pub enum PostType {
+	Blog,
+	Thread,
+}
+
+pub type Topic = Vec<u8>;
+
+pub type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::Balance;
+
+decl_event!(
+	pub enum Event<T> where AccountId = <T as frame_system::Trait>::AccountId {
+		PostCreated(AccountId, Topic, PostType),
+		PostDeleted(AccountId, Topic, PostType),
+	}
+);
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Comment {
+		/// Blogs: User -> Topic -> Message
+		Blog get(fn blog): double_map hasher(twox_64_concat) T::AccountId, hasher(blake2_128_concat) Topic
+			=> Option<Post<BalanceOf<T>, T::BlockNumber>>;
+
+		/// Threads: Topic -> User -> Message
+		Thread get(fn thread): double_map hasher(blake2_128_concat) Topic, hasher(twox_64_concat) T::AccountId
+			=> Option<Post<BalanceOf<T>, T::BlockNumber>>;
+	}
+}
+
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		/// The comment you are looking for does not exist.
+		NotFound,
+		/// The topic is too long.
+		TopicLength,
+		/// The post is too long.
+		PostLength,
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		type Error = Error<T>;
+
+		fn deposit_event() = default;
+
+		#[weight = T::WeightInfo::post()]
+		fn post(origin, post_type: PostType, topic: Topic, post: Vec<u8>) {
+			let poster = ensure_signed(origin)?;
+
+			let topic_len = topic.len();
+			ensure!(topic_len <= T::MaxTopicLength::get(), Error::<T>::TopicLength);
+			let post_len = post.len();
+			ensure!(post_len <= T::MaxPostLength::get(), Error::<T>::PostLength);
+
+			// Deposit is at least MinDeposit, or grows with topic and post length.
+			let deposit = T::ByteDeposit::get().saturating_mul(
```

Comment:
I think `ByteDeposit * (topic_len + post_len) + BaseDeposit` makes more sense over  `max(ByteDeposit * (topic_len + post_len), MinDeposit)`?

So the formula is more similar to other bytes deposit like treasury tip reason